src/main/java/com/cleanroommc/groovyscript/compat/mods/prodigytech/ExplosionFurnace.java

```diff
@@ -0,0 +1,121 @@
+package com.cleanroommc.groovyscript.compat.mods.prodigytech;
+
+import com.cleanroommc.groovyscript.api.GroovyLog;
+import com.cleanroommc.groovyscript.api.IIngredient;
+import com.cleanroommc.groovyscript.api.documentation.annotations.*;
+import com.cleanroommc.groovyscript.compat.mods.ModSupport;
+import com.cleanroommc.groovyscript.helper.SimpleObjectStream;
+import com.cleanroommc.groovyscript.helper.recipe.AbstractRecipeBuilder;
+import com.cleanroommc.groovyscript.registry.VirtualizedRegistry;
+import lykrast.prodigytech.common.recipe.ExplosionFurnaceManager;
+import net.minecraft.item.ItemStack;
+import org.jetbrains.annotations.Nullable;
+
+@RegistryDescription
+public class ExplosionFurnace extends VirtualizedRegistry<ExplosionFurnaceManager.ExplosionFurnaceRecipe> {
+    @RecipeBuilderDescription(example = {
+        @Example(".input(ore('ingotGold'), item('minecraft:diamond')).craftPerReagent(8).power(160).output(item('minecraft:emerald_block'))"),
+        @Example(".input(item('minecraft:stone')).power(160).output(item('minecraft:glowstone'))")
+    })
+    public ExplosionFurnace.RecipeBuilder recipeBuilder() {
+        return new ExplosionFurnace.RecipeBuilder();
+    }
+
+    @Override
+    public void onReload() {
+        removeScripted().forEach(this::remove);
+        restoreFromBackup().forEach(ExplosionFurnaceManager::addRecipe);
+    }
+
+    private boolean remove(ExplosionFurnaceManager.ExplosionFurnaceRecipe recipe) {
```

Comment:
this is used for both `streamRecipes` and `removeScripted().forEach` call this method. this is an issue, since `streamRecipes` should put the recipes it removes in the backups list, and `removeScripted().forEach` should definitely not do that.
i would suggest changing the method to include `addBackup`, and just copying the code into the `forEach` on `removeScripted()`